File: examples/javascript/memory-read-workflow.mjs
import { memside } from "./memside-client.mjs";

let memoryId = process.argv[2];
if (!memoryId) {
  const memories = await memside("/memories?limit=1");
  memoryId = memories[0]?.id;
}

if (!memoryId) {
  console.log("No accessible memories found.");
  process.exit(0);
}

const encodedId = encodeURIComponent(memoryId);
const [memory, batch, revisions, contextMap, subjects] = await Promise.all([
  memside(`/memories/${encodedId}`),
  memside(`/memories/batch?memory_ids=${encodedId}`),
  memside(`/memories/${encodedId}/revisions`),
  memside(`/memories/${encodedId}/context-map`),
  memside(`/memories/${encodedId}/subjects`),
]);

console.log(
  JSON.stringify({ memory, batch, revisions, contextMap, subjects }, null, 2)
);

File: examples/python/memory_read_workflow.py
import json
import sys
from urllib.parse import quote

from memside_http import api_request


memory_id = sys.argv[1] if len(sys.argv) > 1 else None
if not memory_id:
    memories = api_request("GET", "/memories", query={"limit": 1})
    memory_id = memories[0]["id"] if memories else None

if not memory_id:
    print("No accessible memories found.")
    raise SystemExit(0)

encoded_id = quote(memory_id, safe="")
result = {
    "memory": api_request("GET", f"/memories/{encoded_id}"),
    "batch": api_request(
        "GET",
        "/memories/batch",
        query={"memory_ids": memory_id},
    ),
    "revisions": api_request(
        "GET",
        f"/memories/{encoded_id}/revisions",
    ),
    "context_map": api_request(
        "GET",
        f"/memories/{encoded_id}/context-map",
    ),
    "subjects": api_request(
        "GET",
        f"/memories/{encoded_id}/subjects",
    ),
}

print(json.dumps(result, indent=2))
